multi/imxrt-multi/uart.c

```diff
@@ -87,7 +87,17 @@ static int uart_handleIntr(unsigned int n, void *arg)
 {
 	uart_t *uart = (uart_t *)arg;
 
-	*(uart->base + ctrlr) &= ~((1 << 23) | (1 << 21));
+	*(uart->base + ctrlr) &= ~((1 << 27) | (1 << 26) | (1 << 25) | (1 << 23) | (1 << 21));
+
+	/* RX overrun - invalidate fifo and clear status flag */
+	if (*(uart->base + statr) & (1 << 19)) {
+		*(uart->base + fifor) |= (1 << 14);
+		*(uart->base + statr) |= (1 << 19);
+	}
+
+	/* clear RX framing and noise error */
+	if (*(uart->base + statr) & (1 << 18) | (1 << 17))
+		*(uart->base + statr) |= (1 << 18) | (1 << 17);
 
 	return 1;
```

Comment:
Maybe we should wake up interrupt thread if we handled an error (return -1 that case)?

Reply:
What about flushing the RX FIFO and then return -1, in the quoted above case FIFO still contains data.